libs/native/src/main/java/org/elasticsearch/nativeaccess/NativeAccessHolder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess;
+
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.nativeaccess.lib.NativeLibraryProvider;
+
+class NativeAccessHolder {
+
+    protected static final Logger logger = LogManager.getLogger(NativeAccess.class);
+
+    static final NativeAccess INSTANCE;
+
+    static {
+        var libProvider = NativeLibraryProvider.instance();
+        var os = System.getProperty("os.name");
+
+        AbstractNativeAccess inst = null;
+        try {
+            if (os.startsWith("Linux")) {
+                inst = new LinuxNativeAccess(libProvider);
+            } else if (os.startsWith("Mac OS")) {
+                inst = new MacNativeAccess(libProvider);
+            } else if (os.startsWith("Windows")) {
+                inst = new WindowsNativeAccess(libProvider);
+            } else {
+                logger.warn("Unsupported OS [" + os + "]. Native methods will be disabled.");
+            }
+        } catch (LinkageError e) {
+            logger.warn("Unable to load native provider. Native methods will be disabled.", e);
+        }
+        if (inst == null) {
+            inst = new NoopNativeAccess();
```

Comment:
Is this safe going forward? Seems that being lenient here, and then returning a fake implemtation is likely to cause issues. For the existing functionality of determining if the user is root, perhaps defaulting to "false" in this noop implementation is ok but what about for other functionality? I'm thinking about things like the upcoming zstd compression for stored fields. If we can't load a native provider then we have to simply fail. There is no fallback in that scenario. This is going to apply to any scenario where we are bundling native binaries for supported platforms. Once we introduce such a thing, if we are not on an explicitly supported platform, we have to fail to start or risk broken behavior.

Reply:
I matched the existing behavior. If we want to be strict on a particular new call we add, we can do so in the noop impl. I'm happy to be more strict, but currently we are lenient for local development (there have historically been a number of reasons jna might fail to load when trying out ES locally). We do, however, have bootstrap checks which ensure some of these actually executed (eg syscall filtering).

Reply:
Yeah, so I guess the noop impl would just throw an `IllegalOperationException` or something which presumably would cause functionality to not work. We might want to revisit this if we start using native implementations of more critical functionality. It may not make sense going forward to even attempt to start if we can't load the native libraries.

Reply:
I agree, I would prefer not to start at all if we fail to link native access. I'd just like to leave that for a future discussion since this new native library is not changing the current behavior.